Allow for further configuration of psiCollectCron

jonathantredway opened the pull request:
closes #695

patrickhulce commented on packages/server/src/cron/psi-collect.js:
```diff
@@ -17,10 +18,20 @@ const {normalizeCronSchedule} = require('./utils');
+    projectSlug,
+    numberOfRuns = 5,
+    maxNumberOfParallelUrls = Infinity,
+    categories = ['performance', 'accessibility', 'best-practices', 'pwa', 'seo'],
```
maybe don't default it here, so it can fallback to the default in psi-client?

patrickhulce commented on docs/configuration.md:
```diff
@@ -1000,6 +1000,18 @@ _Optional_ The human friendly label for this set of URLs to use when logging sta
 
+##### `psiCollectCron.sites[i].maxNumberOfParallelUrls`
+
+_Optional_ The maximum number of requests to send to the pagespeed insights api concurrently. Defaults to `Infinity` (all urls sent in parallel).
```
```suggestion
_Optional_ The maximum number of requests to send to the PageSpeed Insights API concurrently. Defaults to `Infinity` (all urls sent in parallel).
```

patrickhulce commented on docs/configuration.md:
```diff
@@ -1000,6 +1000,18 @@ _Optional_ The human friendly label for this set of URLs to use when logging sta
+
+##### `psiCollectCron.sites[i].strategy`
+
+_Optional_ The strategy that the pagespeed insights api should use when testing each url in this site. Can be either `desktop` or `mobile`. Defaults to `mobile`.
```
```suggestion
_Optional_ The strategy that the PageSpeed Insights API should use when testing each url in this site. Can be either `desktop` or `mobile`. Defaults to `mobile`.
```

patrickhulce commented on packages/utils/src/psi-runner.js:
```diff
@@ -19,20 +19,22 @@ class PsiRunner {
   /**
    * @param {string} url
-   * @param {{psiApiKey?: string, psiApiEndpoint?: string, psiStrategy?: 'mobile'|'desktop'}} [options]
+   * @param {{psiApiKey?: string, psiApiEndpoint?: string, psiStrategy?: 'mobile'|'desktop', categories?: Array<'performance' | 'accessibility' | 'best-practices' | 'pwa' | 'seo'>}} [options]
```
these options are all redundantly prefixed with `psi` because the CLI `collect` command can also use them, I think it might be worth doing at this level for `psiCategories` as well :) (but in `psi-client`, definitely leave without prefix)

WDYT?

jonathantredway replied:
Yeah, sounds very reasonable to stay consistent.

patrickhulce reviewed (commented):
thanks @jonathantredway this is great!

jonathantredway commented:
@patrickhulce Addressed the feedback ✅, but used the github client to auto-commit your suggestions so it looks like you'll just need to appease googlebot with a consent comment.

patrickhulce commented:
@googlebot I consent

patrickhulce reviewed (approved):
excellent work, thanks @jonathantredway !
